File: billing/src/utils/cors.ts
import { NextApiRequest, NextApiResponse } from "next";

export const cors =
  (fn: (req: NextApiRequest, res: NextApiResponse) => Promise<void>) =>
  async (req: NextApiRequest, res: NextApiResponse) => {
    res.setHeader("Access-Control-Allow-Credentials", "true");
    res.setHeader("Access-Control-Allow-Origin", "https://dashboard.lepton.ai");
    res.setHeader(
      "Access-Control-Allow-Methods",
      "GET,OPTIONS,PATCH,DELETE,POST,PUT",
    );
    res.setHeader(
      "Access-Control-Allow-Headers",
      "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version",
    );
    if (req.method === "OPTIONS") {
      res.status(200).end();
      return;
    }
    return await fn(req, res);
  };

File: lepton-api-server/util/cors.go
package util

import (
	"net/http"

	"github.com/leptonai/lepton/go-pkg/k8s/ingress"
)

// SetCORSForDashboard sets the CORS headers for the response for dashboard.lepton.ai.
func SetCORSForDashboard(h http.Header) {
	// Add the CORS headers after the request is handled to avoid duplication.
	h.Set("Access-Control-Allow-Credentials", "true")
	h.Set("Access-Control-Allow-Origin", "https://dashboard.lepton.ai")
	h.Set("Access-Control-Allow-Methods",
		"POST, PUT, HEAD, PATCH, GET, DELETE, OPTIONS")
	h.Set("Access-Control-Allow-Headers",
		"X-CSRF-Token, X-Requested-With, Accept, Accept-Version, "+
			"Content-Length, Content-MD5, Content-Type, Date, X-Api-Version, "+
			ingress.HTTPHeaderNameForAuthorization+", "+
			ingress.HTTPHeaderNameForDeployment)
}
